cf-execd/cf-execd-runagent.c

```diff
@@ -53,8 +53,16 @@ void HandleRunagentRequest(int conn_fd)
         int length = snprintf(error_buf, sizeof(error_buf),
                               "{\"error\": \"Failed to create file stream from the connection file descriptor: %s\"}",
                               GetErrorStr());
-        assert(length < sizeof(error_buf));
-        write(conn_fd, error_buf, MIN(sizeof(error_buf), (size_t) length));
+        if (length < 0)
+        {
+            Log(LOG_LEVEL_ERR, "Unexpected failure from snprintf while writing error message to connection file descriptor: %s",
+                GetErrorStr());
```

Comment:
This still needs to `write()` a valid JSON with `"error": "something"`.

Reply:
But I don't think `snprintf()` can actually ever return a value less than 0.

> If an output error is encountered, a negative value is returned.

`snprintf()` does no output.